docs/src/main/paradox/jdbc.md

```diff
@@ -1,8 +1,225 @@
+
+There is also an async variant `JdbcProjection.groupedWithinAsync` to which you can pass an async `Handler` that returns a @scala[`Future`]@java[`ComplettionStage`]. Such an async handler doesn't store the projected result in the database, but it integrates with something else (eg: an Actor or some external system). Note that in this case the grouped projection has `at-least-once` semantics. 
+
+## Handler
+
+It's in the @apidoc[JdbcHandler] that you implement the processing of each envelope. It's essentially a consumer function
+from `(JdbcSession, Envelope)` to @scala[`Unit`]@java[`void`]. 
+
+A handler that is consuming `ShoppingCart.Event` from `eventsByTag` can look like this:
+
+Scala
+:  @@snip [JdbcProjectionDocExample.scala](/examples/src/test/scala/docs/jdbc/JdbcProjectionDocExample.scala) { #handler-imports #handler }
+
+Java
+:  @@snip [JdbcProjectionDocExample.java](/examples/src/test/java/jdocs/jdbc/JdbcProjectionDocExample.java) { #handler }
+
+where the `OrderRepository` is an implementation of:
+
+Scala
+:  @@snip [JdbcProjectionDocExample.scala](/examples/src/test/scala/docs/jdbc/JdbcProjectionDocExample.scala) { #repository }
+
+Java
+:  @@snip [JdbcProjectionDocExample.java](/examples/src/test/java/jdocs/jdbc/JdbcProjectionDocExample.java) { #repository }
+
+
+Such simple handlers can also be defined as plain functions via the helper @scala[`JdbcHandler.apply`]@java[`JdbcHandler.fromFunction`] factory method.
+
+### Grouped handler
+
+When using @ref:[`JdbcProjection.groupedWithin`](#groupedwithin) the handler is processing a @scala[`Seq`]@java[`List`] of envelopes.
+
+
+Scala
+:  @@snip [JdbcProjectionDocExample.scala](/examples/src/test/scala/docs/jdbc/JdbcProjectionDocExample.scala) { #handler-imports  #grouped-handler }
+
+Java
+:  @@snip [JdbcProjectionDocExample.java](/examples/src/test/java/jdocs/jdbc/JdbcProjectionDocExample.java) { #grouped-handler }
+
+### Stateful handler
+
+The `JdbcHandler` can be stateful, with variables and mutable data structures. It is invoked by the `Projection` machinery
+one envelope at a time and visibility guarantees between the invocations are handled automatically, i.e. no volatile
+or other concurrency primitives are needed for managing the state as long as it's not accessed by other threads
+than the one that called `process`.
+
+@@@ note
+
+It is important that the `Handler` instance is not shared between several `Projection` instances,
+because then it would be invoked concurrently, which is not how it is intended to be used. Each `Projection`
+instance should use a new `Handler` instance.  
+
+@@@
+
+### Handler lifecycle
+
+You can override the `start` and `stop` methods of the @apidoc[JdbcHandler] to implement initialization
+before first envelope is processed and resource cleanup when the projection is stopped.
+Those methods are also called when the `Projection` is restarted after failure.
+
+See also @ref:[error handling](error.md).
+
+## Processing with Akka Streams
```

Comment:
and this can be removed. referenced to flow.md above. The example is not really showing anything different for Slick, JDBC, Cassandra so I think we can share same example